### ISSUE TITLE
The model is MTR, and the training and testing sets are nuscenes. The problem of reproducing experimental results

### ISSUE THREAD

**Issue by 33zzstar** [vita-epfl/UniTraj#11]
I hope this message finds you well. First, I would like to express my sincere appreciation for your inspiring work. Your research has been truly enlightening and has provided me with valuable insights.

However, I have encountered some difficulties while attempting to reproduce the results presented in your paper. Despite my efforts to follow the instructions and make necessary adjustments based on the issues raised on GitHub, I have not been able to achieve the same outcomes as reported in your work.

Specifically, I modified the configuration file as follows:
`past_len: 21 # history trajectory length, 2.1s
future_len: 60 # future trajectory length, 6s
object_type: [ 'VEHICLE'] # object types included in the training set
line_type: [ 'lane','road_edge','road_line'] # line type to be considered in the input
masked_attributes: [ 'z_axis' ] # attributes to be masked in the input
trajectory_sample_interval: 1 # sample interval for the trajectory
only_train_on_ego: False # only train on AV
center_offset_of_map: [ 30.0, 0.0 ] # center offset of the map
use_cache: False # use cache for data loading
overwrite_cache: False # overwrite existing cache
store_data_in_memory: False # store data in memory`
Additionally, due to the limitations of my GPU, I had to reduce the batch sizes in the train.py file:
`train_batch_size = 4
eval_batch_size = 8`
Apart from these changes, I have not modified any other parts of the code. However, after running the tests, the results I obtained are as follows:
`epoch,step,val/agent_types/vehicle_brier_fde,val/agent_types/vehicle_minADE6,val/agent_types/vehicle_minFDE6,val/agent_types/vehicle_miss_rate,val/brier_fde,val/kalman/easy_brier_fde,val/kalman/easy_minADE6,val/kalman/easy_minFDE6,val/kalman/easy_miss_rate,val/kalman/medium_brier_fde,val/kalman/medium_minADE6,val/kalman/medium_minFDE6,val/kalman/medium_miss_rate,val/minADE6,val/minFDE6,val/miss_rate,val/nuscenes/brier_fde,val/nuscenes/minADE6,val/nuscenes/minFDE6,val/nuscenes/miss_rate,val/traj_type/left_turn_brier_fde,val/traj_type/left_turn_minADE6,val/traj_type/left_turn_minFDE6,val/traj_type/left_turn_miss_rate,val/traj_type/right_turn_brier_fde,val/traj_type/right_turn_minADE6,val/traj_type/right_turn_minFDE6,val/traj_type/right_turn_miss_rate,val/traj_type/stationary_brier_fde,val/traj_type/stationary_minADE6,val/traj_type/stationary_minFDE6,val/traj_type/stationary_miss_rate,val/traj_type/straight_brier_fde,val/traj_type/straight_left_brier_fde,val/traj_type/straight_left_minADE6,val/traj_type/straight_left_minFDE6,val/traj_type/straight_left_miss_rate,val/traj_type/straight_minADE6,val/traj_type/straight_minFDE6,val/traj_type/straight_miss_rate,val/traj_type/straight_right_brier_fde,val/traj_type/straight_right_minADE6,val/traj_type/straight_right_minFDE6,val/traj_type/straight_right_miss_rate
0,0,8.40788745880127,3.1724624633789062,7.57551908493042,0.7599999904632568,8.40788745880127,8.410243034362793,3.1834118366241455,7.567032337188721,0.7551020383834839,8.29244327545166,2.635962963104248,7.991420269012451,1.0,3.1724624633789062,7.57551908493042,0.7599999904632568,8.40788745880127,3.1724624633789062,7.57551908493042,0.7599999904632568,9.51375961303711,2.768587350845337,8.682286262512207,1.0,10.951298713684082,3.1348865032196045,10.291637420654297,1.0,2.257995843887329,0.6625788807868958,1.3122466802597046,0.0,9.097993850708008,6.927713394165039,2.2084782123565674,6.031667709350586,0.8888888955116272,3.6429731845855713,8.272659301757812,0.7809523940086365,6.950369834899902,2.379350185394287,6.136747360229492,1.0`
These results differ from the ones reported in your paper. Moreover, even without modifying the test files, I am still unable to reproduce the same numbers as presented in your work.

I would greatly appreciate any guidance or suggestions you could provide regarding this matter. Is there something else I should modify or consider to successfully reproduce your results?

Thank you in advance for your time and assistance. I look forward to your response and the opportunity to learn from your expertise.

**Comment by Alan-LanFeng** [comment]
May I ask which dataset you are using?

**Comment by 33zzstar** [comment]
I greatly appreciate you taking the time to respond to my inquiry. Your work on investigating the generalization of trajectory prediction is truly fascinating and has piqued my interest.

To address your question regarding the dataset I employed, allow me to provide you with the details. Firstly, I processed the nuscenes dataset v1.0_trainval using scenarionet, which yielded 850 scenes for training purposes. Similarly, I utilized scenarionet to process the nuscenes dataset v1.0_test, resulting in 150 scenes for validation.

Additionally, I made the necessary adjustments to the directory paths within the configuration file to reflect the processed dataset, as follows:



I hope this information can give you a clear understanding of the dataset and its usage in the project. If you have any further inquiries or need further clarification, please feel free to contact us at any time.

Thank you again for your reply and sharing.



&nbsp;


赵梓杉
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年7月3日(星期三) 下午3:40
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [vita-epfl/UniTraj] The model is MTR, and the training and testing sets are nuscenes. The problem of reproducing experimental results (Issue #11)














 
May I ask which dataset you are using?

**Comment by Alan-LanFeng** [comment]
The dataset split you are using is incorrect. When using scenarionet, set --split to the following options: prediction_split = ["mini_train", "mini_val", "train", "train_val", "val"]. This should yield 30000 data for training and 9k data for validation

**Comment by 33zzstar** [comment]
Thank you. Based on your prompt, I have corrected the error.
